### ISSUE TITLE
`bob run` pre+post scripts

### ISSUE THREAD

**Issue by Equanox** [benchkram/bob#32]

**Comment by puengel** [comment]
### Run Workflow
## Assumption
**Pre** can be handled by dependencies. --> No need for further implementation.

**Post** is meant to be done after a Task has finished. Since our run tasks are meant to stay alive it doesn't make sense to do anything after stopping this task.

**Init** is meant to run as soon as the run task is in a running state (e.g. docker-compose up has all containers in running|healthy state). This is what is meant to be implemented with this ticket.

Leads to following run workflow:

```mermaid
flowchart TD
  A[$:bob run environment] --> B{Has dependencies?};
  B -- Yes --> C[Build dependencies!];
  B -- No --> D[Start Runtask environment!];
  C ----> D[Start Runtask environment!];
  D --> E{Is running?};
  E -- No --> F[Wait];
  F --> E;
  E -- Yes --> G{Has Init Task?};
  G -- No --> H[Done. Wait for SigStop.];
  G -- Yes --> I[Run Init Task!];
  I --> H;
```

**Comment by rdnt** [comment]
A usecase for a post script would be to clean up some files that could have been generated during the run time of an interactive task, e.g. logfiles.

**Comment by Equanox** [comment]
I agree for our current usecase **init** is the behaviour we want. The unclear question is how a init task is triggered?

We made the experience that often script can only be executed after preconditions are fullfilled like a `/health` route returning `200`. Same for database rampup as it takes a while till a database is functional when started detached.

I suggest we have a run task with this semantics.
```

**Comment by Equanox** [comment]
I agree for our current use case init is the behavior we want. What should be the trigger for init to run?

**Executing Init**
This seems to be a good place https://github.com/benchkram/bob/blob/main/pkg/ctl/commander.go#L16 to start the init script. binary&compose start procedure is over. Control is given to the TUI as early as possible to assure output is correctly streamed to the user.

**Working together with the TUI**
  * the init cmd should stream the output correctly to the TUI
  * the init cmd must react correctly to a Stop() signal from the TUI  

From a implementation standpoint it seems to be required that init must fullfill the `Control interface` https://github.com/benchkram/bob/blob/main/pkg/ctl/commander.go#L164 to receive the stop/cancel cmd from the TUI correctly. Though it requires us to reuse the shell from the build tasks... the implementation could be relatively easy as restart is not required and the shell can handle a canceable context, see https://github.com/benchkram/bob/blob/main/bobtask/run.go 

**Run task**
A run-task with init looks like this?
```
run:
  server:
    type: binary
    path: ./build/server
    init: |-
      ./run-database-init.sh
```

In the case of `init:` being used on multiple levels in a pipeline the init cmd's should only be executed when the umbrella task triggers. Could make signal handling much easier.

**Out of scope:**
We made the experience that often script can only be executed after preconditions are fulfilled like a `/health` route returning 200. Same for database ram pup as it takes a while till a database is functional when started detached. Tough i would leave those kind of preconditions checks to the script called with init.

**Comment by Equanox** [comment]
Pre scripts closed by #84
